### PR TITLE
Incr. wildcard expansion limit to 25, throw an exception when limit is exceeded, config changes for asynchbase

### PR DESCRIPTION
**Pull request by bsura**

### CODE REVIEW

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java [hunk 1]
```diff
@@ -194,7 +192,8 @@ public DefaultDiscoveryService(SchemaService schemaService, SystemConfiguration
                                 }
                             } else {
                                 if (queries.size() == HARD_LIMIT) {
```
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java]: Should this check be if (queries.size() >= HARD_LIMIT)

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/MetricResources.java [hunk 2]
```diff
@@ -157,10 +160,15 @@ public Response getAsyncMetricsJSON(@Context HttpServletRequest req,
...
 
         for (String expression : expressions) {
-            List<Metric> metricsForThisExpression = metricService.getMetrics(expression);
-
-            metrics.addAll(metricsForThisExpression);
+        	try {
+        		List<Metric> metricsForThisExpression = metricService.getMetrics(expression); 
+        		metrics.addAll(metricsForThisExpression);
+        	} catch(WildcardExpansionLimitExceededException e) {
+        		metricService.dispose();
```
Comment (dilipdevaraj-sfdc) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/MetricResources.java]: If user has multiple expressions to be evaluated, and first expression results in WildcardExpansionLimitExceededException does it make sense to still proceed with the next set of expressions ?

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java (hunk 1 above)
Comment (bsura) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java]: If you take a closer look, == should suffice.

File: ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/MetricResources.java (hunk 2 above)
Comment (bsura) [inline ArgusWebServices/src/main/java/com/salesforce/dva/argus/ws/resources/MetricResources.java]: Yes. Totally. But we currently do not have a way to let the user know how many expressions succeeded and how many failed. For now, it probably makes sense to fail the entire operation.

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java (hunk 1 above)
Comment (dilipdevaraj-sfdc) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/schema/DefaultDiscoveryService.java]: Agreed

**Pull request merged**